common/esxi_get_hardware_versions.yml

```diff
@@ -24,23 +24,24 @@
     get_hardware_versions: true
   register: get_hardware_versions_result
 
-- block:
+- name: "Set facts of ESXi server supported hardware versions"
+  when:
+    - get_hardware_versions_result is defined
+    - get_hardware_versions_result.instance.supported_hardware_versions is defined
```

Comment:
It seems missing one condition "get_hardware_versions_result.instance is defined". Is it not needed?

Reply:
Actually,  `get_hardware_versions_result is defined` can be removed as well, because when checking a variable's property in a variable is defined or not, ansible will firstly check the variable is defined or not. If it is confusing, I can add `get_hardware_versions_result.instance is defined`.

Reply:
If you have confirmed with it, I'm OK with current status.